packages/saber/vue-renderer/app/helpers/scroll-handler.js

```diff
@@ -0,0 +1,100 @@
+/*
+  modified after Vue Router's scroll util
+  https://github.com/vuejs/vue-router/blob/3c7d8ab20f9c716652e92065767a5a44ffb21c13/src/util/scroll.js
+*/
+
+/**
+ *
+ * @param {object} router Vue Router instance
+ * @param {object} to Destination route
+ * @param {object} from Starting route
+ */
+export default function(router, to, from) {
+  if (!router.app) {
+    return
+  }
+
+  const behavior = router.options.scrollBehavior
+  if (!behavior) {
+    return
+  }
+
+  // wait until re-render finishes before scrolling
+  router.app.$nextTick(() => {
+    const shouldScroll = behavior.call(router, to, from, null)
+
+    if (!shouldScroll) {
+      return
+    }
+
+    if (typeof shouldScroll.then === 'function') {
+      shouldScroll
+        .then(shouldScroll => {
+          scrollToPosition(shouldScroll)
+        })
+        .catch(err => {
+          if (process.env.NODE_ENV !== 'production') {
+            console.error(err)
+          }
+        })
+    } else {
+      scrollToPosition(shouldScroll)
+    }
+  })
+}
+
+function getElementPosition(el, offset) {
+  const docEl = document.documentElement
+  const docRect = docEl.getBoundingClientRect()
+  const elRect = el.getBoundingClientRect()
+  return {
+    x: elRect.left - docRect.left - offset.x,
+    y: elRect.top - docRect.top - offset.y
+  }
+}
+
+function isValidPosition(obj) {
+  return isNumber(obj.x) || isNumber(obj.y)
+}
+
+function normalizePosition(obj) {
+  return {
+    x: isNumber(obj.x) ? obj.x : window.pageXOffset,
+    y: isNumber(obj.y) ? obj.y : window.pageYOffset
+  }
+}
+
+function normalizeOffset(obj) {
+  return {
+    x: isNumber(obj.x) ? obj.x : 0,
+    y: isNumber(obj.y) ? obj.y : 0
+  }
+}
+
+function isNumber(v) {
+  return typeof v === 'number'
```

Comment:
is there any chance that v is NaN?

Reply:
No, not really, unless a custom `scrollBehavior` function would return something bad. Even if it would be `NaN`, `window.scroll(NaN, NaN)` has the same effect as `window.scroll(0, 0)`, so not really a concern.